lua/openingh/utils.lua

```diff
@@ -38,15 +38,38 @@ function M.parse_gh_remote(url)
   -- https://github.com/user_or_org/reponame
   -- https://gitlab.com/user_or_org/group/reponame
   -- git@some.github.com:user_or_org/reponame.git
+  -- git@work:user_or_org/reponame.git
   -- ssh://git@some.github.com/user_or_org/reponame.git
   -- ssh://org-12345@some.github.com/org/reponame.git
-  local matches = { string.find(url, "^.+[@/]([%w%.]+%.%w+)[/:](.+)/(%S+)") }
+
+  -- pattern reference: https://www.lua.org/manual/5.4/manual.html#6.4.1
+  local protocol = "ssh"
+  local pattern = "^[^@]+@([^:]+):(.+)/(%S+)"
+  if string.find(url, "^https?://") then
+    protocol = "http"
+    pattern = "^https?://([^/]+)/(.+)/(%S+)"
+  elseif string.find(url, "^ssh://") then
+    protocol = "ssh"
+    pattern = "^ssh://[^@]+@([^/]+)/(.+)/(%S+)"
+  end
```

Comment:
Some notes on the patterns (let me know if these should be comments):
```
^     matches beginning of the string
[^@]+ matches one or more characters that are not "@"
[^:]+ matches one or more characters that are not ":"
(.+)  matches any character one or more times
%S+   matches one or more non-whitespace characters
```
Some spacing may bring clarity into how the patterns work
```
  git   @ github.com : user_or_org / reponame.git
^ [^@]+ @ ([^:]+)    : (.+)        / (%S+)

  https://  github.com / user_or_org / reponame
^ https?:// ([^/]+)    / (.+)        / (%S+)

  ssh:// git   @ github.com / user_or_org / reponame.git
^ ssh:// [^@]+ @ ([^/]+)    / (.+)        / (%S+)
```

Reply:
thank u for your PR, yes please add these comments

Reply:
Done 👍